### DISCUSSION TITLE
Migrate from Ext4 to XFS

### DISCUSSION THREAD

**Discussion by m-roshani** [minio/minio#13999] [Q&A] [answered]
Hi,
Our configuration:
16 servers: 
8 MinIO servers each with 8 drives as Master servers.
8 MinIO servers each with 8 drivers as Slave servers.
MinIO version: 2021-03-04T00:53:13Z
One-way replication has been setup from Master to Slave servers.
Filesystems are Ext4 where we are running out of inodes. 
Based on MinIO calculator, we have fault tolerance of losing up to 2 servers. 

What's your recommendation to migrate to XFS?

Is it possible to do it this way:

1. In one of the Master servers, create XFS partition(s) with the same size as existing drives. 
2. Stopping MinIO service (I suppose there would be no problem as long as requests are redirected to other servers).
3. Unmount Ext4 and mounting XFS partitions in the same drive paths.
4. Starting MinIO service.
5. MinIO would start to use the new XFS filesystems (by its healing mechanism?).
6. When the server is fully operational, do the same with other servers. 

After all, is that OK to have MinIO servers with different underlying filesystems? I mean XFS and Ext4. 

Thank you.

**Comment by harshavardhana** [comment, marked as answer]
> Hi, Our configuration: 16 servers: 8 MinIO servers each with 8 drives as Master servers. 8 MinIO servers each with 8 drivers as Slave servers. MinIO version: 2021-03-04T00:53:13Z One-way replication has been setup from Master to Slave servers. Filesystems are Ext4 where we are running out of inodes. Based on MinIO calculator, we have fault tolerance of losing up to 2 servers.
> 
> What's your recommendation to migrate to XFS?
> 
> Is it possible to do it this way:
> 
> 1. In one of the Master servers, create XFS partition(s) with the same size as existing drives.
> 2. Stopping MinIO service (I suppose there would be no problem as long as requests are redirected to other servers).
> 3. Unmount Ext4 and mounting XFS partitions in the same drive paths.
> 4. Starting MinIO service.
> 5. MinIO would start to use the new XFS filesystems (by its healing mechanism?).
> 6. When the server is fully operational, do the same with other servers.
> 
> After all, is that OK to have MinIO servers with different underlying filesystems? I mean XFS and Ext4.
> 
> Thank you.

Please upgrade first to latest release and once that is done we can move to discussing migration.
